### ISSUE TITLE
inf norm inconsistent with numpy

### ISSUE THREAD

**Issue by ailzhang** [pytorch/pytorch#27241]
If you add `inf, -inf` into the list https://github.com/pytorch/pytorch/blob/master/test/test_torch.py#L9274 it'll fail the test. 

So from numpy doc, `inf` norm is defined as `max(sum(abs(x), axis=1))` for matrix and `max(abs(x))` for vector, but it seems we always implemented `max(abs(x))`. 

short repro:

```
In [1]: import torch

In [2]: import numpy as np

In [3]: a = torch.rand(3, 3)

In [4]: b = a.numpy()

In [5]: import math

In [6]: torch.norm(a, math.inf)
Out[6]: tensor(0.7553)

In [7]: np.linalg.norm(b, math.inf)
Out[7]: 1.4294269
```

**Comment by SsnL** [comment]
duplicate of https://github.com/pytorch/pytorch/issues/24802

**Comment by ailzhang** [comment]
Oh thanks @SsnL for pointing it out! Closing in favor of the other issue.
